### ISSUE TITLE
Alias /?postcode=CV15RR to /postcode/CV15RR ??

### ISSUE THREAD

**Issue by JoeMitchell** [DemocracyClub/UK-Polling-Stations#524]
Request from Coventry:

>If you can get it so that https://wheredoivote.co.uk/embed/?postcode=CV15RR can redirect to https://wheredoivote.co.uk/embed/postcode/CV15RR/ then I can get a widget to work that I can share across everyone using Jadu as their CMS.

Sounds cool!

**Comment by chris48s** [comment]
1. We can probably add an alias so that `https://wheredoivote.co.uk/embed/?postcode={postcode}` behaves like `https://wheredoivote.co.uk/embed/postcode/{postcode}/` but what happens in the case where the postcode is ambiguous and the user needs to choose an address. For example: https://wheredoivote.co.uk/address_select/CV67JF/

Have they considered that/does that break their plan?

2. Another approach would be that if they want to build their own CMS widget, they could use the beta API to do that, but I'm concerned that it may be a bit _too_ beta/subject to change at this stage. Depends where we've got to with EC/Squiz..

Perhaps as a first thing you could pose the first question and see where we go - if it isn't an issue, happy days. If it is, we can catch up on the second bit when we chat on Wednesday. Also if you don't want to be a middleman in this conversation, feel free to link your contact at Coventry to this thread to discuss.

Ta

**Comment by cvaho171** [comment]
> We can probably add an alias so that https://wheredoivote.co.uk/embed/?postcode={postcode} behaves like https://wheredoivote.co.uk/embed/postcode/{postcode}/

If https://wheredoivote.co.uk/embed/?postcode={postcode} would redirect to https://wheredoivote.co.uk/embed/postcode/{postcode}/ I can't see why split postcodes would be an issue? Wouldn't the same page appear, so they get to choose an address or are shown their polling station?

**Comment by chris48s** [comment]
Hi @cvaho171

> I can't see why split postcodes would be an issue?

So the reason I ask the question is because if it is a problem for you that https://wheredoivote.co.uk/ submits to https://wheredoivote.co.uk/postcode/CV67JF/ I wanted to check in case it might also be problematic that https://wheredoivote.co.uk/address_select/CV67JF/ then submits to https://wheredoivote.co.uk/address/e08000026-5702-534-sewall-highway-cv67jf/ (as opposed to, for example https://wheredoivote.co.uk/address?slug=e08000026-5702-534-sewall-highway-cv67jf or whatever).

Given it isn't, happy days - that makes things easier :)

I've submitted PR #533 which allows the `/postcode` endpoint to optionally accept postcode as a GET param - so once that is deployed, you'll be able to call `https://wheredoivote.co.uk/embed/postcode?postcode=mypostcode`.

We'll update you once that change has been reviewed and deployed

**Comment by cvaho171** [comment]
That seems fine to me, if I'm understanding it correctly: for every postcode they'll have to pick their full address from the picker?

I'm happy as long as they get the correct polling station at the end of it.

Thanks!

**Comment by chris48s** [comment]
@cvaho171
This is now deployed, so you can call (for example)
https://wheredoivote.co.uk/embed/postcode/?postcode=CV21PN
Cheers

**Comment by cvaho171** [comment]
@chris48s Thank you. It's now on http://www.coventry.gov.uk/pollingstations.

**Comment by chris48s** [comment]
@cvaho171

Thanks very much for doing this. Having seen what you've done with it, I've spotted that there is some validation/cleanup of postcodes that we do in our input form which is now bypassed when someone else's form submits to our site. I've raised an issue for this here https://github.com/DemocracyClub/UK-Polling-Stations/issues/608 and I will aim to have a look at adding the missing validation/cleanup later this week.


A couple of bits of feedback:

* Your form allows users to submit with a blank postcode. You might want to consider making postcode a required field.
* We mainly intended https://wheredoivote.co.uk/embed/ to be used when councils are embedding our site in an `<iframe>` (hence having no headers and footers). Given you are just directing the user to our site (rather then embedding in an `<iframe>`) it might look nicer to submit to https://wheredoivote.co.uk/postcode/?postcode=CV32DF - but I can also see that you might prefer the 'unbranded' look.

I'll leave those with you.

Cheers :)
